### PR TITLE
Add validation prior to running security tools

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments
Adds validation prior to running security tools.  Required because some security tools will throw an exception if you pass it invalid inputs.

## Pull Request Checklist
Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:
 - [X] Ensure that `make build` completes successfully
 - [X] Rebase your branch against the latest commit of the target branch, which likely should be `master`
 - [X] If there is an issue associated with your Pull Request, align your PR title with [this documentation](https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: functions [hunk 1]
```diff
@@ -29,61 +29,93 @@ function _feedback() {
...
+    # If the initialization was already run successfully in an non-interactive
+    # shell, don't run it again
+    if [[ -r "/terraform_init_complete" ]]; then
+      _feedback INFO "Skipping `terraform init` because it was already run on $(date -d @"$(cat /terraform_init_complete)")"
+    else
+      command terraform init &>/terraform_init_output
+      process_command_exit_status "${?}" "terraform init" "initialization"
+      return=${?}
+      if [[ ${return:-1} != 0 ]]; then
+        cat /terraform_init_output
```
Comment (pb-seiso) [inline functions:79]: See above comment.

File: functions [hunk 2]
```diff
@@ -29,61 +29,93 @@ function _feedback() {
...
+        return "${return}"
+      fi
+    fi
+    # If the initialization was already run successfully in an non-interactive
+    # shell, don't run it again
+    if [[ -r "/terraform_validate_complete" ]]; then
+      _feedback INFO "Skipping `terraform validate` because it was already run on $(date -d @"$(cat /terraform_validate_complete)")"
+    else
+      command terraform validate &>/terraform_validate_output
```
Comment (pb-seiso) [inline functions:88]: Similar to the above comments, `terraform_validate_output` -> `terraform_validate_complete`

File: functions [hunk 3]
```diff
@@ -29,61 +29,93 @@ function _feedback() {
...
+  elif [[ "${!tfsec_env_var,,}" == "true" ]]; then
     _feedback WARNING "Skipping tfsec due to the ${tfsec_env_var} environment variable value"
   # If the security scan was already run successfully in an non-interactive
   # shell, don't run it again
   elif [[ -r "/tfsec_complete" ]]; then
-    _feedback INFO "Skipping tfsec because it was already run on $(date -d @$(cat /tfsec_complete))"
+    _feedback INFO "Skipping tfsec because it was already run on $(date -d @"$(cat /tfsec_complete)")"
   # Otherwise, run the security tool
   else
     tfsec . &>/tfsec_output
```
Comment (pb-seiso) [inline functions:117]: `tfsec_output` --> `tfsec_complete`

File: functions [hunk 4]
```diff
@@ -29,61 +29,93 @@ function _feedback() {
...
-    process_security_tool_exit_status "${?}" "tfsec" "recursive scan"
+    process_command_exit_status "${?}" "tfsec" "recursive scan"
     return=${?}
     if [[ ${return:-1} != 0 ]]; then
       cat /tfsec_output
```
Comment (pb-seiso) [inline functions:121]: `tfsec_output` --> `tfsec_complete`

**Comment by JonZeolla** [comment]
@pb-seiso I don't see an issue; perhaps there's a misunderstanding of expectations for interactive vs non-interactive use.  Can you share the test you were doing?

File: functions [hunk 5]
```diff
@@ -29,61 +29,93 @@ function _feedback() {
...
     command "${FUNCNAME[0]}" "$@"
     return $?
+  else
+    ## Validate the input prior to running security tooling
+    # If the initialization was already run successfully in an non-interactive
+    # shell, don't run it again
+    if [[ -r "/terraform_init_complete" ]]; then
+      _feedback INFO "Skipping `terraform init` because it was already run on $(date -d @"$(cat /terraform_init_complete)")"
+    else
+      command terraform init &>/terraform_init_output
```
[the remark this answers is not in the document]
Comment (pb-seiso) [reply functions:75]: Closing this - misunderstood what the intention was here.

**Comment by pb-seiso** [comment]
> @pb-seiso I don't see an issue; perhaps there's a misunderstanding of expectations for interactive vs non-interactive use. Can you share the test you were doing?

Yeah that was a misunderstanding on my part, thanks for clarifying.

**Comment by JonZeolla** [comment]
The last two force pushes were:
1. Drop the commit for updating deps.  Now that I found the root cause it is unneeded and we can do that separately.
2. Rebase on `master`.

**Review by pb-seiso — APPROVED** [review]
LGTM!

**Pull request merged**